solutions/USACO feb18-teleportation.mdx

```diff
@@ -0,0 +1,49 @@
+---
+id: usaco-807
+source: USACO Bronze 2018 February
+title: Teleportation
+author: Ananth Kashyap
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_teleport_bronze_feb18.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(1)$
+
+<LanguageSection>
+
+<PySection>
+
+```python
+# Take in data using Python file i/o system
+file_in = open('teleport.in')
+data = file_in.read().strip().split('\n')
+a, b, x, y = map(int, data[0].split(' '))
+
+# Manually sort a, b to be increasing order
+if a > b:
+	a, b = b, a
+
+# Manually sort x, y to be increasing order
+if x > y:
+	x, y = y, x
+
+# Set base distance as distance needed to travel without using teleporter
+base_distance = abs(a - b)
+
+# Set teleporter distance to be the minimal travel distance using teleporter
+# He must go from a to x as otherwise, the teleporter would move him backwards
+# He must then move from point y to point b
```

Comment:
```suggestion
# The teleporter distance is the minimal travel distance using a teleporter, which is the sum of the abs(a - x) and abs(b - y)
```

Reply:
make sure that the lines are less than or equal to 80 characters; breaking the line at the comma would probably work

Reply:
Feels a bit excessive (again it's just comments, not a full solution writeup)

Possibly
```suggestion
# Set teleporter distance to be the travel distance using the teleporter
```
suffices